plan/preprocess.go

```diff
@@ -48,6 +48,8 @@ type preprocessor struct {
 	/* For Select Statement. */
 	// table map to lookup and check table name conflict.
 	tableMap map[string]int
+	// table map to lookup and check derived-table(subselect) name conflict.
+	derivedTableMap map[string]int
```

Comment:
Is the `int` map values is meaningless ? If so, how about `derivedTableSet map[string]struct{}` ?

Reply:
The int value is useful.